src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,160 @@
+package nextstep.subway.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import nextstep.subway.exception.AlreadyRegisterStationException;
+
+@Embeddable
+public class Sections {
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST,
+            CascadeType.MERGE}, orphanRemoval = true)
+    private final List<Section> sections = new ArrayList<>();
+
+    public void addSection(Section section) {
+        if (sections.isEmpty()) {
+            sections.add(section);
+            return;
+        }
+
+        validateRegisterStation(section.getUpStation(), section.getDownStation());
+
+        if (insertSectionIsUpStation(section)) {
+            return;
+        }
+
+        if (insertSectionIsDownStation(section)) {
+            return;
+        }
+
+        if (insertSectionBetweenSections(section)) {
+            return;
+        }
+
+        throw new IllegalArgumentException("section 을 넣을 수 없습니다.");
+    }
+
+    private void validateRegisterStation(Station upStation, Station downStation) {
+        boolean containUpStation = false;
+        boolean containDownStation = false;
+
+        for (Section section : sections) {
+            if (section.getUpStation().equals(upStation) || section.getDownStation().equals(upStation)) {
+                containUpStation = true;
+            }
+            if (section.getUpStation().equals(downStation) || section.getDownStation().equals(downStation)) {
+                containDownStation = true;
+            }
+        }
+        if (containUpStation && containDownStation) {
```

Comment:
위의 containUpStation, containDownStation를 제거하시고 아래와 같이 section 객체에 메서드를 추가해보면 어떨까요

```suggestion
        if (section.containStation(upStation) && section.containStation(downStation)) {
```